src/main/java/network/brightspots/rcv/CommonDataFormatReader.java

```diff
@@ -118,12 +198,204 @@
     return rankings;
   }
 
-  // parse the given file into a List of CastVoteRecords for tabulation
-  void parseCvrFile(List<CastVoteRecord> castVoteRecords) throws UnrecognizedCandidatesException {
+  // helper to extract the required CVRContest from the input CVR
+  CVRContest getCvrContest(CVR cvr, Contest contestToTabulate) {
+    CVRContest cvrContestToTabulate = null;
+    // find current snapshot
+    CVRSnapshot currentSnapshot = null;
+    for (CVRSnapshot cvrSnapshot : cvr.CVRSnapshot) {
+      if (cvrSnapshot.ObjectId.equals(cvr.CurrentSnapshotId)) {
+        currentSnapshot = cvrSnapshot;
+        break;
+      }
+    }
+    // find CVRContest which matches the Contest we are tabulating
+    for (CVRContest cvrContest : currentSnapshot.CVRContest) {
```

Comment:
This can throw an NPE, so should handle safely with an assert, if statement, etc.